src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -260,6 +261,132 @@ public static void BuildChain_WithCertificatePolicy_NoMatch()
+            using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+            {
+                // Check that microsoft.com's root certificate IS installed in the machine root store as a sanity step
+                using (var machineRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
+                {
+                    machineRootStore.Open(OpenFlags.ReadOnly);
+                    bool foundCert = false;
+
+                    foreach (var machineCert in machineRootStore.Certificates)
+                    {
+                        if (machineCert.Equals(microsoftDotComRoot))
+                        {
+                            foundCert = true;
+                        }
+
+                        machineCert.Dispose();
+
+                        if (foundCert)
+                        {
+                            break;
+                        }
+                    }
+
+                    Assert.True(foundCert, string.Format("Did not find expected certificate with thumbprint '{0}' in the machine root store", microsoftDotComRoot.Thumbprint));
+                }
+
+                // Concievably at this point there could still be something wrong and we still don't chain build correctly - if that's 
+                // the case, then there's likely something wrong with the machine. Validating that happy path is out of scope 
+                // of this particular test - instead, use BuildChain_MicrosoftDotCom_WithSystemRootCerts as a canary 
+
+
+                // Check that microsoft.com's root certificate is NOT installed on in the user cert store as a sanity step
+                // We won't try to install the microsoft.com root cert into the user root store if it's already there
+                using (var userRootStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                {
+                    userRootStore.Open(OpenFlags.ReadOnly);
+
+                    foreach (var userCert in userRootStore.Certificates)
+                    {
+                        bool foundCert = false;
+                        if (userCert.Equals(microsoftDotComRoot))
+                        {
+                            foundCert = true;
+                        }
+
+                        userCert.Dispose();
+
+                        if (foundCert)
+                        {
+                            shouldInstallCertToUserStore = false;
+                            break;
+                        }
+                    }
+                }
+
+                using (var userRootStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                using (var chainHolder = new ChainHolder())
+                {
+                    try
+                    {
+                        if (shouldInstallCertToUserStore)
+                        {
+                            userRootStore.Add(microsoftDotComRoot); // throws CryptographicException
```

Comment:
I'm not sure what the comment means here.   But you didn't call userRootStore.Open(OpenMode.ReadWrite), if that helps... :)